src/components/Analytics/genaralCharts/CorrectResponsesPercentage.js

```diff
@@ -0,0 +1,116 @@
+import Box from '@mui/material/Box';
+
+import { ANALYTICS_GENERAL_CORRECT_RESPONSE_PERCENTAGE_CY } from '../../../config/selectors';
+import {
+  defaultLayout,
+  defaultSettings,
+  hoverData,
+} from '../../../utils/plotUtils';
+import { computeCorrectness, getDataWithId } from '../../context/utilities';
+
+const Plot = createPlotlyComponent(Plotly);
+
+/**
+ * Component that renders the percentage of correct responses in the quiz
+ *
+ * @param maxWidth the maximum width of the chart
+ * @param goToDetailedQuestion The callback to call to be redirected to the detailed chart corresponding to
+ * the question that has been clicked on
+ * @param responses The responses provided by the user to the quiz
+ * @param order The order in which the questions appear in the quiz
+ * @param questions The question for which to display detailed information
+ */
+const CorrectResponsesPercentage = ({
+  maxWidth,
+  goToDetailedQuestion,
+  responses,
+  order,
+  questions,
+}) => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+
+  const responsesByQId = useMemo(
+    () => responses.groupBy((r) => r.data.questionId),
+    [responses]
+  );
+
+  const chartData = useMemo(() => {
+    return order.reduce(
+      (acc, qId, idx) => {
+        const question = getDataWithId(questions, qId);
+        const responses = responsesByQId.get(qId);
+        const nbCorrect = responses.reduce(
+          (acc, next) =>
+            computeCorrectness(next.data, question.data) ? acc + 1 : acc,
+          0
+        );
+
+        return {
+          data: {
+            x: [...acc.data.x, `Q${idx + 1}`],
+            y: [...acc.data.y, nbCorrect / responses.size],
+          },
+          hoverText: [...acc.hoverText, question.data.question],
+          qIds: [...acc.qIds, question.id],
+        };
+      },
+      {
+        data: { x: [], y: [] },
+        hoverText: [],
+        qIds: [],
+      }
+    );
+  }, [questions, responsesByQId, order]);
+
+  return (
+    <Box
+      sx={{ width: '100%' }}
+      data-cy={ANALYTICS_GENERAL_CORRECT_RESPONSE_PERCENTAGE_CY}
+    >
+      <Plot
+        data={[
+          {
+            name: t('Quiz correct response percentage'),
+            type: 'scatter',
+            ...chartData.data,
+            marker: {
+              color: theme.palette.primary.main,
+            },
+            ...hoverData(
+              chartData.hoverText,
+              chartData.qIds,
+              `%{hovertext}<br><br> - ${t(
+                'Percentage correct responses'
+              )}: %{y:.1%} <extra></extra>`,
+              theme.palette.primary.main
+            ),
+            fill: 'tozeroy',
+          },
+        ]}
+        layout={{
+          ...defaultLayout({
+            title: t('Quiz correct response percentage'),
+            width: maxWidth,
+            percentage: true,
+            maxValueY: 1.1,
+          }),
+          xaxis: {
+            range: [-0.2, order.length - 0.8],
```

Comment:
As otherwise the dot that marks the value would appear as a half dot. So I wanted the range to be `0.2` bigger at the beginning of the chart, and at the end.

I added a comment to explain it.